src/models/ili9342c.rs

```diff
@@ -0,0 +1,193 @@
+            None => write_command(di, Instruction::SWRESET, &[])?,
+        }
+
+        delay.delay_us(120_000);
+
+        write_command(di, Instruction::COLMOD, &[0b0101_0101])?; // 16bit 65k colors
+
+        init_common(di, delay, options).map_err(|_| Error::DisplayError)
+    }
+
+    fn write_pixels<DI, I>(&mut self, di: &mut DI, colors: I) -> Result<(), DisplayError>
+    where
+        DI: WriteOnlyDataCommand,
+        I: IntoIterator<Item = Self::ColorFormat>,
+    {
+        write_command(di, Instruction::RAMWR, &[])?;
+        let mut iter = colors.into_iter().map(|c| c.into_storage());
+
+        let buf = DataFormat::U16BEIter(&mut iter);
+        di.send_data(buf)
+    }
+
+    fn display_size(&self, orientation: Orientation) -> (u16, u16) {
+        match orientation {
+            Orientation::Portrait(_) | Orientation::PortraitInverted(_) => (320, 240),
+            Orientation::Landscape(_) | Orientation::LandscapeInverted(_) => (240, 320),
+        }
+    }
+}
+
+impl Model for ILI9342CRgb666 {
+    type ColorFormat = Rgb666;
+
+    fn new() -> Self {
+        Self
+    }
+
+    fn init<RST, DELAY, DI>(
+        &mut self,
+        di: &mut DI,
+        rst: &mut Option<RST>,
+        delay: &mut DELAY,
+        options: DisplayOptions,
+    ) -> Result<u8, Error<RST::Error>>
+    where
+        RST: OutputPin,
+        DELAY: DelayUs<u32>,
+        DI: WriteOnlyDataCommand,
+    {
+        match rst {
+            Some(ref mut rst) => self.hard_reset(rst, delay)?,
+            None => write_command(di, Instruction::SWRESET, &[])?,
+        }
+
+        delay.delay_us(120_000);
+
+        write_command(di, Instruction::COLMOD, &[0b0110_0110])?; // 18bit 262k colors
+
+        init_common(di, delay, options).map_err(|_| Error::DisplayError)
+    }
+
+    fn write_pixels<DI, I>(&mut self, di: &mut DI, colors: I) -> Result<(), DisplayError>
+    where
+        DI: WriteOnlyDataCommand,
+        I: IntoIterator<Item = Self::ColorFormat>,
+    {
+        write_command(di, Instruction::RAMWR, &[])?;
+        let mut iter = colors.into_iter().flat_map(|c| {
+            let red = c.r() << 2;
+            let green = c.g() << 2;
+            let blue = c.b() << 2;
+            [red, green, blue]
+        });
+
+        let buf = DataFormat::U8Iter(&mut iter);
+        di.send_data(buf)
+    }
+
+    fn display_size(&self, orientation: Orientation) -> (u16, u16) {
+        match orientation {
+            Orientation::Portrait(_) | Orientation::PortraitInverted(_) => (320, 240),
+            Orientation::Landscape(_) | Orientation::LandscapeInverted(_) => (240, 320),
+        }
+    }
+}
+
+// simplified constructor for Display
+
+impl<DI, RST> Display<DI, RST, ILI9342CRgb565>
+where
+    DI: WriteOnlyDataCommand,
+    RST: OutputPin,
+{
+    ///
+    /// Creates a new [Display] instance with [ILI9342C] as the [Model]
+    /// *WARNING* Rgb565 only works on non-SPI setups with the ILI9342C!
```

Comment:
Your pull request description mentions using this on an ESP32-S3-BOX but I also see that you were only able to use 565 and not 666. I thought the box was using SPI - is this documentation that 565 only works with non-SPI setups correct?

Reply:
No, the docs are wrong. The ILI934X controllers support RGB565 and RGB666 via SPI. The comment was probably copied from the ILI9486 support, where it would be correct.